server/routers/interview.ts

```diff
@@ -1,55 +1,80 @@
-import * as z from 'zod';
+/* eslint-disable local-rules/require-data-mapper */
 import { prisma } from '~/utils/db';
 import { publicProcedure, protectedProcedure, router } from '~/server/trpc';
-import { safeLoader } from '~/utils/safeLoader';
 
-const deleteSingleInterviewSchema = z.object({
-  id: z.string(),
-});
+import {
+  interviewIdSchema,
+  interviewListInputSchema,
+  participantIdentifierSchema,
+} from '~/shared/schemas';
 
-const deleteManyInterviewsSchema = z.array(
-  z.object({
-    id: z.string(),
-  }),
-);
+export const interviewRouter = router({
+  create: publicProcedure
+    .input(participantIdentifierSchema)
+    .mutation(async ({ input: identifier }) => {
+      try {
+        // get the active protocol id to connect to the interview
+        const activeProtocol = await prisma.protocol.findFirst({
+          where: { active: true },
+        });
 
-const InterviewValidation = z.array(
-  z.object({
-    id: z.string(),
-    startTime: z.date(),
-    finishTime: z.date().nullable(),
-    exportTime: z.date().nullable(),
-    lastUpdated: z.date(),
-    participantId: z.string(),
-    protocolId: z.string(),
-    currentStep: z.number(),
-    participant: z.object({
-      id: z.string(),
-      identifier: z.string(),
+        if (!activeProtocol) {
+          return {
+            error: 'Failed to create interview: no active protocol',
+            createdInterview: null,
+          };
+        }
+
+        const createdInterview = await prisma.interview.create({
+          data: {
+            startTime: new Date(),
+            lastUpdated: new Date(),
+            currentStep: 0,
+            network: '',
+            participant: {
+              create: {
+                identifier: identifier,
+              },
+            },
+            protocol: {
+              connect: {
+                id: activeProtocol.id,
+              },
+            },
+          },
+        });
+
+        return { error: null, createdInterview };
+      } catch (error) {
+        return {
+          error: 'Failed to create interview',
+          createdInterview: null,
+        };
+      }
     }),
-    protocol: z.object({
-      id: z.string(),
-      name: z.string(),
+  get: router({
+    all: publicProcedure.query(async () => {
+      const interviews = await prisma.interview.findMany({
+        include: {
+          protocol: true,
```

Comment:
I think this also needs to include the participant, since this is used primarily by the interview dashboard route to render the table. 

We should think about if it makes sense to configure what is included in the query input. 

Actually on that point, we need to think about pagination and whatnot with these large queries. I'll add a clickup task to cover that.

Reply:
included participant here [0ea46cc](https://github.com/complexdatacollective/Fresco/pull/20/commits/0ea46cca62dc0de59eeb44ff7b3264c910a23955)